mysql/resource_binlog.go

```diff
@@ -0,0 +1,129 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceBinLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateBinLog,
+		UpdateContext: UpdateBinLog,
+		ReadContext:   ReadBinLog,
+		DeleteContext: DeleteBinLog,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportDatabase,
+		},
+		Schema: map[string]*schema.Schema{
+			"retention_period": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NULL",
+			},
+		},
+	}
+}
+
+func CreateBinLog(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stmtSQL := binlogConfigSQL(d)
+	log.Println("Executing statement:", stmtSQL)
+
+	_, err = db.ExecContext(ctx, stmtSQL)
+	if err != nil {
+		return diag.Errorf("failed running SQL to set binlog retention period: %v", err)
+	}
+
+	d.SetId(d.Get("retention_period").(string))
```

Comment:
>  does changing the retention_period cause the existing resource to be deleted and a new created rather than updating?

The resource id never gets changed, but the id is tied to the initial retention_period which doesn't make a how lot of sense if it changes.

Reply:
Right, It didn't make sense to relate the retention period with the ID, indeed. I created an ID unrelated to the retention time period.